warp/Network/Wai/Handler/Warp/Header.hs

```diff
@@ -19,27 +19,35 @@ indexRequestHeader hdr = traverseHeader hdr requestMaxIndex requestKeyIndex
 
 idxContentLength,idxTransferEncoding,idxExpect :: Int
 idxConnection,idxRange,idxHost :: Int
-idxContentLength    = 0
```

Comment:
OK. I confirmed. Consider the following code:

```haskell
data X = A | B | C | D | E deriving Enum

main :: IO ()
main = print $ 1234 + fromEnum E
```

`ghc -O -ddump-prep` generates 1238 (ie 1234 + 4)!